preview/FS-1068-open-type-declaration.md

```diff
@@ -0,0 +1,273 @@
+# F# RFC FS-1068 - Open Type Declaration
+
+The design suggestion [Open static classes](https://github.com/fsharp/fslang-suggestions/issues/383) has been marked "approved in principle".
+This RFC covers the detailed proposal for this suggestion.
+
+
+[Discussion thread](https://github.com/fsharp/fslang-design/issues/352)
+
+* [x] Approved in principle
+* [x] Details: [Resolved to Preview](https://github.com/fsharp/fslang-design/issues/352)
+* [x] Implementation: [Complete to Preview](https://github.com/dotnet/fsharp/pull/6325)
+
+
+## Summary
+[summary]: #summary
+
+Add support for opening types to gain access to static members and nested types, e.g. `open type System.Math`. For example
+
+```fsharp
+> open type System.Math;;
+> Min(1.0, 2.0);;
+val it : float = 1.0
+
+> Min(2.0, 1.0);;
+val it : float = 1.0
+```
+
+## Motivation
+[motivation]: #motivation
+
+This greatly increases the expressivity of F# DSLs by allowing method-API facilities such as named arguments, optional arguments and type-directed overloading to be used in the DSL design.
+
+Type providers can provide static classes, hence this would allow type providers to provide "unqualified" names. This useful for some projections where the natural thing is unqualified, e.g. the "R" type provider always required `R.plot` etc. With this feature, `R.plot` can now be `plot`.
+
+Additionally, important C# DSL APIs are starting to appear that effectively require this.
+
+## Detailed design
+[design]: #detailed-design
+
+Opening of types allows for treating a type somewhat as if it were a module with module-bound functions inside of it. For example:
+
+```fsharp
+type A() =
+    static member M(x: int) = x * 2
+
+open type A
+
+M(2)
+```
+
+However, **members are not functions**, so very different rules apply beyond simple cases like the above. This primarily concerns name resolution when overloads are involved, or when considering whether or not we resolve a method or a property first.
```

Comment:
> However, **members are not functions**, so very different rules apply beyond simple cases like the above. This primarily concerns name resolution when overloads are involved, or when considering whether or not we resolve a method or a property first.

Neither of these are related to module-bound functions as far as I can tell. You make mention of rules, including the application of some rules in the previous example, but there is no mention of rules for resolving either a module-bound function or a member.

Additionally, what are the concerns? I don't see any description about how we resolve overloads or the resolution of a name of a method or a property. There is an alternatives section describing the construction of a method overload group, but what does it contrast with?